src/Product/ProductSyncer.php

```diff
@@ -357,10 +356,25 @@ protected function handle_delete_errors( array $invalid_products ) {
 	 * @throws ProductSyncerException If the Google Merchant Center connection is not ready.
 	 */
 	protected function validate_merchant_center_setup(): void {
-		if ( ! $this->merchant_center->is_ready_for_syncing() ) {
+		if ( ! $this->merchant_center->is_ready() ) {
 			do_action( 'woocommerce_gla_error', 'Cannot sync any products before setting up Google Merchant Center.', __METHOD__ );
 
 			throw new ProductSyncerException( __( 'Google Merchant Center has not been set up correctly. Please review your configuration.', 'google-listings-and-ads' ) );
 		}
+
+		if ( ! $this->merchant_center->should_sync() ) {
```

Comment:
Also what do you think if we add some tests for this?

Reply:
The tests are added in 

ProductSyncerTest - https://github.com/woocommerce/google-listings-and-ads/pull/2326/files#diff-e1a250ff800d7687a728767bcc4817c6ca40f2fa9e97d9846e4b967ff338ee1fR551

CouponSyncerTest - https://github.com/woocommerce/google-listings-and-ads/pull/2326/files#diff-19117240c30f979181c4332c987f76f2ab19a5fd4b460d3345c8e68d195a1116R149

Reply:
See my other comments about the tests.